slack_connection.go

```diff
@@ -0,0 +1,128 @@
+	SourceType string `json:"source_type"`
+
+	ChannelID        string `json:"channel_id"`
+	ChannelName      string `json:"channel_name"`
+	NotificationType string `json:"notification_type"`
+
+	Config SlackConnectionConfig `json:"config"`
+}
+
+// SlackConnectionObject is an API object returned by getter functions.
+type SlackConnectionObject struct {
+	SlackConnection
+	APIObject
+}
+
+// ListSlackConnectionsResponse is an API object returned by the list function.
+type ListSlackConnectionsResponse struct {
+	Connections []SlackConnectionObject `json:"slack_connections"`
+	APIListObject
+}
+
+// ListSlackConnectionsOptions is the data structure used when calling the ListSlackConnections API endpoint.
+type ListSlackConnectionsOptions struct {
+	// Limit is the pagination parameter that limits the number of results per
+	// page. PagerDuty defaults this value to 50 if omitted, and sets an upper
+	// bound of 100.
+	Limit uint `url:"limit,omitempty"`
+
+	// Offset is the pagination parameter that specifies the offset at which to
+	// start pagination results. When trying to request the next page of
+	// results, the new Offset value should be currentOffset + Limit.
+	Offset uint `url:"offset,omitempty"`
+}
+
+// CreateSlackConnectionWithContext creates a Slack connection.
+func (c *Client) CreateSlackConnectionWithContext(ctx context.Context, slackTeamID string, s SlackConnection) (SlackConnectionObject, error) {
+	d := map[string]SlackConnection{
+		"slack_connection": s,
+	}
+
+	resp, err := c.post(ctx, "/integration-slack/workspaces/"+slackTeamID+"/connections", d, nil)
+	return getSlackConnectionFromResponse(c, resp, err)
+}
+
+// GetSlackConnection gets a Slack connection.
+func (c *Client) GetSlackConnectionWithContext(ctx context.Context, slackTeamID, connectionID string) (SlackConnectionObject, error) {
+	resp, err := c.get(ctx, "/integration-slack/workspaces/"+slackTeamID+"/connections/"+connectionID)
+	return getSlackConnectionFromResponse(c, resp, err)
+}
+
+// DeleteSlackConnectionWithContext deletes a Slack connection.
+func (c *Client) DeleteSlackConnectionWithContext(ctx context.Context, slackTeamID, connectionID string) error {
+	_, err := c.delete(ctx, "/integration-slack/workspaces/"+slackTeamID+"/connections/"+connectionID)
+	return err
+}
+
+// UpdateSlackConnectionWithContext updates an existing Slack connection.
+func (c *Client) UpdateSlackConnectionWithContext(ctx context.Context, slackTeamID, connectionID string, s SlackConnection) (SlackConnectionObject, error) {
+	d := map[string]SlackConnection{
+		"slack_connection": s,
+	}
+
+	resp, err := c.put(ctx, "/integration-slack/workspaces/"+slackTeamID+"/connections/"+connectionID, d, nil)
+	return getSlackConnectionFromResponse(c, resp, err)
+}
+
+// ListSlackConnectionsWithContext lists Slack connections.
+func (c *Client) ListSlackConnectionsWithContext(ctx context.Context, slackTeamID string, o ListSlackConnectionsOptions) (*ListSlackConnectionsResponse, error) {
```

Comment:
Please remove `WithContext` from the name, since there is not a non-context version. Also, I don't think this should return a pointer type. We're going to be moving away from those when it makes sense (#270).